### PR TITLE
Fix the flash of unstyled content on all browsers

### PR DESCRIPTION
**Pull request by bkase**
Namely Firefox is the problem now (but it turns out opera would have
done this too if it still existed).

Tested by cmd+shift+r-ing on chrome and firefox (a lot). Compared to staging, it's way better. See code comments for explanations of the fixes.

### CODE REVIEW

File: frontend/website/src/Style.re [hunk 1]
```diff
@@ -54,60 +59,138 @@ module Typeface = {
...
+      };
+
+    let string_of_fontStyle =
+      fun
+      | `normal => "normal"
+      | `italic => "italic"
+      | `oblique => "oblique"
+      | `initial => "initial"
+      | `inherit_ => "inherit"
+      | `unset => "unset";
```
Comment (Schmavery) [inline frontend/website/src/Style.re]: Can we remove this because our font style is always normal

File: frontend/website/src/Style.re [hunk 2]
```diff
@@ -54,60 +59,138 @@ module Typeface = {
...
+      | `medium => "500"
+      | `semiBold => "600"
+      | `bold => "700"
+      | `extraBold => "800"
+      | `black => "900"
+      | `lighter => "lighter"
+      | `bolder => "bolder"
+      | `initial => "initial"
+      | `inherit_ => "inherit"
+      | `unset => "unset"
```
Comment (Schmavery) [inline frontend/website/src/Style.re]: We can remove the unused cases

File: frontend/website/src/Style.re [hunk 3]
```diff
@@ -54,60 +59,138 @@ module Typeface = {
...
+
+      let fontStyle =
+        Belt.Option.mapWithDefault(fontStyle, "", s => "font-style: " ++ s);
+      let fontWeight =
+        Belt.Option.mapWithDefault(fontWeight, "", w =>
+          "font-weight: " ++ string_of_fontWeight(w)
+        );
+      let asString = {j|@font-face {
+      font-family: $fontFamily;
+      src: $src;
```
Comment (Schmavery) [inline frontend/website/src/Style.re]: Is it worth adding support for format("woff") etc now that we control this?

File: frontend/website/src/Style.re [hunk 4]
```diff
@@ -40,6 +40,11 @@ module Colors = {
   let midnight = `rgb((31, 45, 61));
 };
 
+module Emotion = {
+  [@bs.module "emotion"]
+  external rawInjectGlobal: string => unit = "injectGlobal";
+};
```
Comment (Schmavery) [inline frontend/website/src/Style.re]: Is this used still?

File: frontend/website/src/Style.re (hunk 3 above)
Comment (bkase) [reply frontend/website/src/Style.re]: already done (see the string constructed there)

File: frontend/website/src/Style.re [hunk 5]
```diff
@@ -42,72 +42,125 @@ module Colors = {
...
-    fontFamily(
-      fontFace(
-        ~fontFamily="IBM Plex Serif",
-        ~src=[
-          url("/static/font/IBMPlexSerif-Medium-Latin1.woff2"),
-          url("/static/font/IBMPlexSerif-Medium-Latin1.woff"),
+  // To prevent "flash of unstyled text" on some browsers (firefox), we need
+  // to do insane things to mitigate it. Even though the CSS working group
+  // created `font-display: block` for this purpose, Firefox chooses to not
+  // follow the standard "wait for 3seconds before showing fallback fonts."
```
Comment (cmr) [inline frontend/website/src/Style.re]: According to the code it does: https://dxr.mozilla.org/mozilla-central/source/layout/style/nsFontFaceLoader.cpp#82

But in my firefox I definitely don't see _any_ block period whatsoever.

Comment (bkase) [reply frontend/website/src/Style.re]: Exactly

Comment (bkase) [reply frontend/website/src/Style.re]: Maybe it's fixed on master

**Pull request merged**